test/tools/ossfuzz/SolidityGenerator.cpp

```diff
@@ -0,0 +1,32 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#include <test/tools/ossfuzz/SolidityGenerator.h>
+
+#include <libsolutil/Whiskers.h>
+
+using namespace solidity::test::fuzzer;
+using namespace solidity::util;
+using namespace std;
+
+string SolidityGenerator::generateTestProgram()
```

Comment:
So this PR is establishing the framework? Given this generated starting contract doesn't seem to be very exciting.

Reply:
Yea I think that's the idea

Reply:
Yeah, this PR is setting up the framework. Code for Test case generation will follow in upcoming PRs

Reply:
Can you add a comment here explaining what this does and what needs to be done? And what is blocking this PR?

Reply:
Done

Reply:
Shouldn't this have an input, like the seed? Or how will this generate code?

Reply:
The idea is to write a generator based on the antlr4 grammar here. https://github.com/ethereum/solidity/blob/develop/docs/grammar/Solidity.g4

Starting point is a test unit that contains one or more source units. Each source unit contains one or more elements such as function, contract etc.

The input to the generator is a deterministic seed (`uint64_t`) passed here

https://github.com/ethereum/solidity/pull/9722/files#diff-3f64bed29063599c885c0e99d8aa0d98734424f0fac0009517c0b04ba0cb0a8fR38

This seed fully determines the sequence of generation steps undertaken by the generator so that a given seed always produces the same test program.

The test program itself is generated by randomly choosing one of N elements from the grammar and recursive descent.

Reply:
The seed is given and used in the constructor